### DISCUSSION TITLE
有人知道 “为 Affix display 附加更多触发事件” 中对 touch* 的监听有什么特别理由吗？

### DISCUSSION THREAD

**Discussion by kabeep** [ant-design/ant-design#44438] [Q&A] [answered]
### Q:
在这次 [Commit: Attach more events for Affix display](https://github.com/ant-design/ant-design/commit/01d4acb7a9b030be3552b4ca19b4e899dc2bb7c7) 中增加了 touchstart, touchmove, touchend 的事件监听器，
作为 _置空 affixStyle, placeholderStyle 状态 & 设置状态为 Prepare_ 事件的触发方式。

它改变了 state 造成 re-render，即使它没有发生实质性变化，这个方式仍然在 5.x 中保留，有人对这样做的理由有想法吗？它解决了移动端的什么问题，这在 Commit 日志中没有显现。

### Reproduce
打开 [codesandbox](https://codesandbox.io/s/affix-wu-yi-yi-chu-fa-re-render-antd-4-24-13-forked-vfwt8w)
=> Open in New Window
=> 打开 **开发者工具** 并使用 **切换设备工具栏** 切换到任意移动设备
=> 在不激活 <Affix /> 的状态下 touchstart/touchmove/touchend
=> re-render Affix

### Screenshot
![image](https://github.com/ant-design/ant-design/assets/56389128/5fd18e45-8919-4eed-9ab3-00cb023d76ef)

**Comment by kabeep** [comment]
@afc163 大佬得闲分享下，是不是和 safari 地址栏导致的 viewport 问题有关

**Comment by afc163** [comment]
这个有年头了，当时没留更多信息，还原不了现场了。

**Reply by yoyo837** [reply]
当前没要求修改都走PR呀 😄

**Reply by afc163** [reply]
是的，当时的提交很不规范。

**Reply by afc163** [reply, marked as answer]
我猜当时有些移动端滚动页面不是用 scrollbar 原生实现的。
